lib/ensure_content_type.js

```diff
@@ -45,19 +52,15 @@ function checkContentType(hyper, req, next, expectedContentType, responsePromise
                     // Don't fail the request.
                     return res;
                 }
+            }
 
-                // Do not re-render if the response was only rendered within
-                // the last two minutes.
-                if (res.headers.etag
-                        && Date.now() - mwUtil.extractDateFromEtag(res.headers.etag) < 120000) {
-                    return res;
-                }
-
+            // The old content might not have any profile at all
+            if (expectedProfile && actualProfile !== expectedProfile) {
```

Comment:
I was assuming that it does not make sense to configure this content-type versioning filter if the spec does not have any versioned content type. If we want to protect against such a misconfiguration, perhaps we could just fatal on startup?